src/core/Form/Checkbox/Checkbox.tsx

```diff
@@ -66,50 +118,136 @@ class DefaultCheckbox extends Component<CheckboxProps> {
+      ('value' in this.props && (typeof value !== 'string' || value === ''))
+    ) {
+      logger.warn(
+        'Name and value props should have non-empty values if provided.',
+      );
+    }
+
+    const statusTextId = `${id}-statusText`;
+    const hintTextId = `${id}-hintText`;
+
+    const getDescribedBy = () => {
+      if (statusText || hintText || ariaDescribedBy) {
+        return {
+          'aria-describedby': [
+            ...(statusText ? [statusTextId] : []),
+            ...(hintText ? [hintTextId] : []),
+            ...(ariaDescribedBy ? [ariaDescribedBy] : []),
+          ].join(' '),
+        };
+      }
+      return {};
+    };
+
+    const newCheckboxInputProps = {
+      disabled,
+      id,
+      'aria-label': ariaLabel,
+      'aria-labelledby': ariaLabelledBy,
+      'aria-invalid': status === 'error',
+      checked: !!checkedState,
+      className: checkboxClassNames.input,
+      onChange: this.handleClick,
+      name,
+      ...getDescribedBy(),
+      ...(value ? { value } : {}),
+    };
+
     return (
-      <StyledCheckbox
-        disabled={disabled}
-        onClick={this.handleClick}
-        {...passProps}
-        status={status}
-        className={classnames(baseClassName, className, {
-          [checkboxClassNames.error]: status === 'error' && !disabled,
-          [checkboxClassNames.checked]: checkedState && !disabled,
-          [checkboxClassNames.large]: variant === 'large',
-          [checkboxClassNames.disabled]: !!disabled,
-        })}
-      >
-        {!!checkedState && (
-          <Icon
-            icon="check"
-            className={classnames(iconBaseClassName, {
-              [iconClassnames.checked]: checkedState && !disabled,
-              [iconClassnames.error]: status === 'error' && !disabled,
-              [iconClassnames.disabled]: !!disabled,
-            })}
-          />
+      <HtmlDiv
+        className={classnames(
+          checkboxClassNames.container,
+          className,
+          baseClassName,
+          {
+            [checkboxClassNames.error]: status === 'error' && !disabled,
+            [checkboxClassNames.checked]: checkedState && !disabled,
+            [checkboxClassNames.large]: variant === 'large',
+            [checkboxClassNames.disabled]: !!disabled,
+          },
         )}
-        {children}
-      </StyledCheckbox>
+      >
+        <HtmlInput {...newCheckboxInputProps} type="checkbox" />
+        <HtmlLabel
+          htmlFor={id}
+          className={checkboxClassNames.label}
+          {...passProps}
+        >
+          {!!checkedState && (
+            <Icon
+              icon="check"
+              className={classnames(iconBaseClassName, {
+                [iconClassnames.checked]: checkedState && !disabled,
+                [iconClassnames.error]: status === 'error' && !disabled,
+                [iconClassnames.disabled]: !!disabled,
+              })}
+            />
+          )}
```

Comment:
Would it be nice if we try to minimise the logic in the "main return"? 

So e.g the logic for showing the icon would be like this:

```
    const CheckedIcon = () => (
      <>
        {!!checkedState && (
          <Icon
            icon="check"
            className={classnames(iconBaseClassName, {
              [iconClassnames.checked]: checkedState && !disabled,
              [iconClassnames.error]: status === 'error' && !disabled,
              [iconClassnames.disabled]: !!disabled,
            })}
          />
        )}
      </>
    );
```

And using it:
```
          <CheckedIcon />
```

Reply:
In this case I'd prefer showing the logic for hiding in the template. If CheckedIcon is introduced inside render then props won't be an issue - not worth moving outside in my opinion. How about:
```
const CheckedIcon = () => (
          <Icon
            icon="check"
            className={classnames(iconBaseClassName, {
              [iconClassnames.checked]: checkedState && !disabled,
              [iconClassnames.error]: status === 'error' && !disabled,
              [iconClassnames.disabled]: !!disabled,
            })}
          />
    );
```
And
`{!!checkedState && (<CheckedIcon />)}`
in the template

Reply:
Yes, that might be better; seeing the reason in the template why it might not be showing (even though it is quite logical from the name of the component.. but still).

Reply:
This is now changed as suggested.